diff --git a/vendor/knative.dev/pkg/webhook/resourcesemantics/conversion/reconciler.go b/vendor/knative.dev/pkg/webhook/resourcesemantics/conversion/reconciler.go
index 103e6a5e..8151849f 100644
--- a/vendor/knative.dev/pkg/webhook/resourcesemantics/conversion/reconciler.go
+++ b/vendor/knative.dev/pkg/webhook/resourcesemantics/conversion/reconciler.go
@@ -19,6 +19,7 @@ package conversion
 import (
 	"context"
 	"fmt"
+	"os"

 	"go.uber.org/zap"
 	apixv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
@@ -75,6 +76,9 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 		return err
 	}

+	if os.Getenv("USE_OLM_TLS") != "" { // olm will do the crd update
+		return nil
+	}
 	cacert, ok := secret.Data[certresources.CACert]
 	if !ok {
 		return fmt.Errorf("secret %q is missing %q key", r.secretName, certresources.CACert)
