diff --git a/vendor/knative.dev/operator/pkg/reconciler/common/ha.go b/vendor/knative.dev/operator/pkg/reconciler/common/ha.go
index 9dbc884f3..221c696ad 100644
--- a/vendor/knative.dev/operator/pkg/reconciler/common/ha.go
+++ b/vendor/knative.dev/operator/pkg/reconciler/common/ha.go
@@ -26,7 +26,7 @@ import (

 func haUnSupported(obj base.KComponent) sets.String {
 	return sets.NewString(
-		"pingsource-mt-adapter",
+		"pingsource-mt-adapter", "imc-controller", "imc-dispatcher", "mt-broker-controller", "mt-broker-filter", "mt-broker-ingress",
 	)
 }
 
