diff --git a/vendor/knative.dev/reconciler-test/pkg/manifest/installer.go b/vendor/knative.dev/reconciler-test/pkg/manifest/installer.go
index add000964..e06db8eee 100644
--- a/vendor/knative.dev/reconciler-test/pkg/manifest/installer.go
+++ b/vendor/knative.dev/reconciler-test/pkg/manifest/installer.go
@@ -22,6 +22,9 @@ import (
 	"encoding/json"
 	"io/fs"
 	"strings"
+	"runtime"
+	"os"
+	"path"
 
 	"k8s.io/client-go/util/retry"
 	"knative.dev/pkg/injection/clients/dynamicclient"
@@ -82,6 +85,17 @@ func InstallYamlFS(ctx context.Context, fsys fs.FS, base map[string]interface{})
 	return manifest, nil
 }
 
+// Deprecated: use InstallYamlFS instead.
+func InstallLocalYaml(ctx context.Context, base map[string]interface{}) (Manifest, error) {
+	log := loggingFrom(ctx, "InstallLocalYaml")
+	pwd, _ := os.Getwd()
+	log.Debug("PWD: ", pwd)
+	_, filename, _, _ := runtime.Caller(1)
+	log.Debug("FILENAME: ", filename)
+
+	return InstallYamlFS(ctx, os.DirFS(path.Dir(filename)), base)
+}
+
 func ImagesFromFS(ctx context.Context, fsys fs.FS) []string {
 	log := logging.FromContext(ctx)
 	var images []string
