diff --git a/vendor/knative.dev/eventing/test/upgrade/prober/receiver.go b/vendor/knative.dev/eventing/test/upgrade/prober/receiver.go
index 763c88635..ca2dcf37c 100644
--- a/vendor/knative.dev/eventing/test/upgrade/prober/receiver.go
+++ b/vendor/knative.dev/eventing/test/upgrade/prober/receiver.go
@@ -93,6 +93,10 @@ func (p *prober) createReceiverDeployment() *appsv1.Deployment {
 					Labels: map[string]string{
 						"app": receiver.Name,
 					},
+					Annotations: map[string]string{
+						"sidecar.istio.io/inject":                "true",
+						"sidecar.istio.io/rewriteAppHTTPProbers": "true",
+					},
 				},
 				Spec: corev1.PodSpec{
 					Volumes: []corev1.Volume{{
diff --git a/vendor/knative.dev/eventing/test/upgrade/prober/sender.go b/vendor/knative.dev/eventing/test/upgrade/prober/sender.go
index 864df0e63..8dbc8304e 100644
--- a/vendor/knative.dev/eventing/test/upgrade/prober/sender.go
+++ b/vendor/knative.dev/eventing/test/upgrade/prober/sender.go
@@ -53,6 +53,10 @@ func (p *prober) deploySender() {
 					Labels: map[string]string{
 						"app": sender.Name,
 					},
+					Annotations: map[string]string{
+						"sidecar.istio.io/inject":                "true",
+						"sidecar.istio.io/rewriteAppHTTPProbers": "true",
+					},
 				},
 				Spec: corev1.PodSpec{
 					Volumes: []corev1.Volume{{
diff --git a/vendor/knative.dev/eventing/test/upgrade/prober/verify.go b/vendor/knative.dev/eventing/test/upgrade/prober/verify.go
index 55dc12d38..66591db9b 100644
--- a/vendor/knative.dev/eventing/test/upgrade/prober/verify.go
+++ b/vendor/knative.dev/eventing/test/upgrade/prober/verify.go
@@ -264,6 +264,10 @@ func (p *prober) deployFetcher() *batchv1.Job {
 					Labels: map[string]string{
 						"app": fetcherName,
 					},
+					Annotations: map[string]string{
+						"sidecar.istio.io/inject":                "true",
+						"sidecar.istio.io/rewriteAppHTTPProbers": "true",
+					},
 				},
 				Spec: corev1.PodSpec{
 					RestartPolicy: corev1.RestartPolicyNever,
