diff --git a/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go b/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go
index 1985d7268..f0d86d1f7 100644
--- a/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go
+++ b/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go
@@ -139,6 +139,7 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ke *v1beta1.KnativeEvent
 			return nil
 		},
 		r.transform,
+		r.handleBackstageResources,
 		r.handleTLSResources,
 		manifests.Install,
 		manifests.SetManifestPaths, // setting path right after applying manifests to populate paths
