diff --git a/vendor/knative.dev/operator/pkg/reconciler/common/images.go b/vendor/knative.dev/operator/pkg/reconciler/common/images.go
index 12ebd53ba..18721e6f8 100644
--- a/vendor/knative.dev/operator/pkg/reconciler/common/images.go
+++ b/vendor/knative.dev/operator/pkg/reconciler/common/images.go
@@ -121,6 +121,7 @@ func ImageTransform(registry *base.Registry, log *zap.SugaredLogger) mf.Transfor
 				env := &container.Env[j]
 				if image, ok := registry.Override[env.Name]; ok {
 					env.Value = image
+					env.ValueFrom = nil
 				}
 			}
 		}
