diff --git a/vendor/knative.dev/operator/pkg/reconciler/common/stages.go b/vendor/knative.dev/operator/pkg/reconciler/common/stages.go
index ce299592a..d6bf5d5b1 100644
--- a/vendor/knative.dev/operator/pkg/reconciler/common/stages.go
+++ b/vendor/knative.dev/operator/pkg/reconciler/common/stages.go
@@ -18,11 +18,15 @@ package common
 
 import (
 	"context"
+	"fmt"
 	"strings"
 
 	mf "github.com/manifestival/manifestival"
-	"knative.dev/operator/pkg/apis/operator/base"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"knative.dev/pkg/logging"
+
+	"knative.dev/operator/pkg/apis/operator/base"
 )
 
 // Stage represents a step in the reconcile process
@@ -113,6 +117,12 @@ func DeleteObsoleteResources(ctx context.Context, instance base.KComponent, fetc
 		return NoOp
 	}
 	return func(_ context.Context, manifest *mf.Manifest, _ base.KComponent) error {
-		return installed.Filter(mf.NoCRDs, mf.Not(mf.In(*manifest))).Delete()
+		for _, r := range installed.Filter(mf.NoCRDs, mf.Not(mf.In(*manifest))).Resources() {
+			m, _ := mf.ManifestFrom(mf.Slice([]unstructured.Unstructured{r}), mf.UseClient(manifest.Client))
+			if err := m.Delete(); err != nil && !meta.IsNoMatchError(err) {
+				return fmt.Errorf("failed to delete obsolete resources: %w", err)
+			}
+		}
+		return nil
 	}
 }
