diff --git a/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/controller.go b/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/controller.go
index 170926033..50c9b28f2 100644
--- a/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/controller.go
+++ b/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/controller.go
@@ -37,6 +37,9 @@ import (
 	"knative.dev/pkg/injection/clients/dynamicclient"
 	"knative.dev/pkg/logging"
 	"k8s.io/apimachinery/pkg/util/wait"
+
+	namespaceinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/namespace"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // NewController initializes the controller and is called by the generated code
@@ -76,6 +79,22 @@ func NewExtendedController(generator common.ExtensionGenerator) injection.Contro
 			FilterFunc: controller.FilterControllerGVK(v1beta1.SchemeGroupVersion.WithKind("KnativeEventing")),
 			Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 		})
+		namespaceinformer.Get(ctx).Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				ns, ok := obj.(metav1.Object)
+				if !ok {
+					return false
+				}
+				v, ok := ns.GetLabels()["kubernetes.io/metadata.name"]
+				if !ok {
+					return false
+				}
+				return v == "knative-eventing"
+			},
+			Handler: controller.HandleAll(func(i interface{}) {
+				impl.GlobalResync(knativeEventingInformer.Informer())
+			}),
+		})
 
 		go func(){
 			err = wait.PollImmediate(3*time.Second, 5*time.Minute, func() (bool, error) {
