diff --git a/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go b/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go
--- a/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go	(revision db82259937f61c191d61a8458c7079da57c7fb60)
+++ b/vendor/knative.dev/operator/pkg/reconciler/knativeeventing/knativeeventing.go	(date 1691045721039)
@@ -119,6 +119,10 @@
 		source.AppendTargetSources,
 		common.AppendAdditionalManifests,
 		r.appendExtensionManifests,
+		func(ctx context.Context, manifest *mf.Manifest, component base.KComponent) error {
+			*manifest = manifest.Filter(mf.Not(mf.All(mf.ByKind("Namespace"), mf.ByName("knative-eventing"))))
+			return nil
+		},
 		r.transform,
 		manifests.Install,
 		common.CheckDeployments,
