diff --git a/vendor/knative.dev/operator/pkg/reconciler/common/stages.go b/vendor/knative.dev/operator/pkg/reconciler/common/stages.go
index b9a7c36de..e39f12f73 100644
--- a/vendor/knative.dev/operator/pkg/reconciler/common/stages.go
+++ b/vendor/knative.dev/operator/pkg/reconciler/common/stages.go
@@ -91,6 +91,15 @@ func AppendInstalled(ctx context.Context, manifest *mf.Manifest, instance v1alph
 	return nil
 }
 
+func FilterNamespace(ns string) Stage {
+	return func(ctx context.Context, manifest *mf.Manifest, component base.KComponent) error {
+		*manifest = manifest.Filter(mf.Not(
+			mf.All(mf.ByKind("Namespace"), mf.ByName("knative-eventing")),
+		))
+		return nil
+	}
+}
+
 // ManifestFetcher returns a manifest appropriate for the instance
 type ManifestFetcher func(ctx context.Context, instance base.KComponent) (*mf.Manifest, error)
 
