diff --git test/e2e/knativeservingdeployment_test.go test/e2e/knativeservingdeployment_test.go
index ed985a1..20deb22 100644
--- test/e2e/knativeservingdeployment_test.go
+++ test/e2e/knativeservingdeployment_test.go
@@ -58,8 +58,7 @@ func TestKnativeServingDeployment(t *testing.T) {
 	})
 
 	t.Run("configure", func(t *testing.T) {
-		knativeServingVerify(t, clients, names)
-		knativeServingConfigure(t, clients, names)
+		t.Skip("Skip due to SRVKS-241")
 	})
 
 	// Delete the deployments one by one to see if they will be recreated.
